astroid/bases.py

```diff
@@ -120,7 +122,11 @@ def infer(self, context=None):
         yield self
 
 
-def _infer_stmts(stmts, context, frame=None):
+def _infer_stmts(
+    stmts: list[InferenceResult],
+    context: InferenceContext | None,
+    frame: nodes.LocalsDictNodeNG | None = None,
+) -> Iterator[InferenceResult]:
```

Comment:
Have been thinking about this again. We currently mix the return types quite a bit. `NodesNG.infer` returns a `Generator` and `NodeNG._infer` an `Interator`. In practice, it doesn't make much of a difference, but it might make sense to at least be consistent.

**Suggestion**
`-> Generator[InferenceResult, None, None]` for all `*infer*` methods / functions. Brains can continue to return an `Iterator`.